rio-window/src/platform_impl/linux/x11/ime/context.rs

```diff
@@ -174,7 +174,7 @@ impl PreeditCallbacks {
         let start_callback = create_xim_callback(client_data, unsafe {
             mem::transmute::<
                 usize,
-                unsafe extern "C" fn(*mut x11_dl::xlib::_XIM, *mut i8, *mut i8),
```

Comment:
it actually should be 

```rust
mem::transmute::<usize, unsafe extern "C" fn(ffi::XIM, ffi::XPointer, ffi::XPointer)>(
                preedit_start_callback as usize,
            )
```

Reply:
need to fix the rest of the foreign functions as well https://github.com/rust-windowing/winit/blob/b674d20edf4c212d4a98040e4a99013a76287f1e/src/platform_impl/linux/x11/ime/context.rs#L4

Reply:
doesn't need to include the `c_char` https://github.com/rust-windowing/winit/blob/b674d20edf4c212d4a98040e4a99013a76287f1e/src/platform_impl/linux/x11/ime/context.rs#L3

Reply:
@orhun lemme know if you would like me to do the changes 🙏